### PR TITLE
[RFR] Implemented the script in Node.js

### PR DESCRIPTION
**Pull request by KittyGiraudel**
- JavaScript is simpler to read than Bash;
- JavaScript is cross-platform;
- It will make it possible to have a Node.js API as well;
- It makes attribute parsing / preserving easier;
- I wanted to do it.

### CODE REVIEW

File: bin/spritesh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env ruby
 
 bin_dir = File.expand_path(File.dirname(__FILE__))
-shell_script_path = File.join(bin_dir, 'sprite.sh')
+shell_script_path = File.join(bin_dir, 'spritesh.js')
```
Comment (KittyGiraudel) [inline bin/spritesh:4]: Not sure how this will play out with the gem. Maybe it’s not necessary anymore.

**Comment by KittyGiraudel** [comment]
Ping @edenspiekermann/js for review.

File: bin/spritesh.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+const program = require('commander');
+const cheerio = require('cheerio');
+
+program
+  .option('-i, --input [input]', 'Specifies input dir (current dir by default)')
+  .option('-o, --output [output]', 'Specifies output file ("./sprite.svg" by default)')
+  .option('-v, --viewbox [viewbox]', 'Specifies viewBox attribute (parsed by default)')
+  .option('-p, --prefix [prefix]', 'Specifies prefix for id attribute (none by default)')
+  .option('-q, --quiet', 'Disable informative output')
+  .parse(process.argv);
```
Comment (eschaefer) [inline bin/spritesh.js:14]: so much cleaner than Bash, wow.

**Comment by KittyGiraudel** [comment]
@cadejscroggins: is there anything specific that should be done now that we will move the script to Node.js instead of Bash, regarding the Gem version?

**Comment by xvvvyz** [comment]
The only issue I see is that this makes node a dependency unless we set it up with a ruby js interpreter. 👎

**Comment by KittyGiraudel** [comment]
How would you tackle this? Is there an example somewhere?

**Comment by xvvvyz** [comment]
Maybe something [like this](https://github.com/sstephenson/execjs) to parse the js script in `bin/spritesh`? Feels rather hacky though.

**Comment by KittyGiraudel** [comment]
There is also: http://mokagio.github.io/tech-journal/2015/02/25/ruby-make-a-gem-that-wraps-a-node-module.html.

**Pull request merged**